src/psbt.cpp

```diff
@@ -267,6 +268,12 @@ void PSBTOutput::Merge(const PSBTOutput& output)
     if (m_tap_internal_key.IsNull() && !output.m_tap_internal_key.IsNull()) m_tap_internal_key = output.m_tap_internal_key;
     if (m_tap_tree.has_value() && !output.m_tap_tree.has_value()) m_tap_tree = output.m_tap_tree;
 }
+
+bool PSBTInputSigned(const PSBTInput& input, const CScript& script_pubkey)
```

Comment:
Considering that a `PSBTInput` already has the utxo, it's unnecessary to have the spk passed in by the caller. It could just be looked up.

Reply:
for non-witness utxos you have the whole txn, so you kind of need the prevout index of that tx? Pushed a change that splits concepts but requires the PSBT/index tuple to be passed in instead.

Reply:
The caller, `SignPSBTInput` fetches the UTXO after it does this check. So that could be reordered to fetch the UTXO first, then pass that into this function. Thins would remove the need for the lookup to be done twice.

This would also be a lot easier with PSBTv2 implemented as the prevout info will be in the `PSBTInput`, regardless of version.